Add document pip support

davideberlein opened the pull request:
- Add support for detecting video element visibility in Document PiP
- Drop opacity check as the intersection observer is not triggered if an element changes opacity and the check is flawed anyway as opacity of parents isn't considered.
- Add toggle document pip button to `examples/demo` page.

Closes #1317

lukasIO reviewed (commented):
This looks great, thank you @davideberlein ! 

For the demo page two small nits: 
It would be nice if the pip toggle could keep track of it's current state (enabled vs disabled) and there are still some exceptions spewing in the console after popping out the window where some updates are still trying to render to the window.document directly instead of to the pip document.

davideberlein commented:
> ions spewing in the console after popping out the window where some updates are still trying to render to the window.document directly instead of to the pip document

Good finding there, I missed 2 stops which I added and cleaned up the rest.

lukasIO reviewed (approved):
looks good, excited to support this new possibility :)
